nmpolicy/internal/parser/parser.go

```diff
@@ -101,21 +97,34 @@ func (p *Parser) currentToken() *lexer.Token {
 	return &p.tokens[p.currentTokenIdx]
 }
 
-func (p *Parser) parseIdentity() ast.Node {
-	return ast.Node{
+func (p *Parser) parseIdentity() {
+	p.lastNode = &ast.Node{
 		Meta:     ast.Meta{Position: p.currentToken().Position},
 		Terminal: ast.Terminal{Identity: &p.currentToken().Literal},
 	}
 }
 
-func (p *Parser) parseString() ast.Node {
-	return ast.Node{
+func (p *Parser) parseString() {
+	p.lastNode = &ast.Node{
 		Meta:     ast.Meta{Position: p.currentToken().Position},
 		Terminal: ast.Terminal{String: &p.currentToken().Literal},
 	}
 }
 
-func (p *Parser) parsePath() (*ast.Node, error) {
+func (p *Parser) parseNumber() error {
```

Comment:
I think you should keep all parser* methods with the same signature. Either they should have an error or not.